satpy/tests/test_composites.py

```diff
@@ -216,17 +221,72 @@ def test_basic_no_sharpen(self):
+        """Test that high resolution green with no neutral band can be passed."""
+        from satpy.composites import RatioSharpenedRGB
+        comp = RatioSharpenedRGB(name='true_color', high_resolution_band='green', neutral_resolution_band=None)
+        res = comp((self.ds1, self.ds2, self.ds3), optional_datasets=(self.ds4,))
+        res = res.values
+        self.assertEqual(res.shape, (3, 2, 2))
+        np.testing.assert_allclose(res[0], np.array([[5/3, 5/3], [np.nan, 0.0]], dtype=np.float64))
+        np.testing.assert_allclose(res[1], np.array([[1.0, 1.0], [np.nan, 1.0]], dtype=np.float64))
+        np.testing.assert_allclose(res[2], np.array([[4/3, 4/3], [np.nan, 4/3]], dtype=np.float64))
+
+    def test_high_green_neutral_blue(self):
+        """Test that high resolution green with blue neutral band can be passed."""
+        from satpy.composites import RatioSharpenedRGB
+        comp = RatioSharpenedRGB(name='true_color', high_resolution_band='green', neutral_resolution_band='blue')
+        res = comp((self.ds1, self.ds2, self.ds3), optional_datasets=(self.ds4,))
+        res = res.values
+        self.assertEqual(res.shape, (3, 2, 2))
+        np.testing.assert_allclose(res[0], np.array([[5/3, 5/3], [np.nan, 0.0]], dtype=np.float64))
+        np.testing.assert_allclose(res[1], np.array([[1.0, 1.0], [np.nan, 1.0]], dtype=np.float64))
+        np.testing.assert_allclose(res[2], np.array([[4.0, 4.0], [np.nan, 4.0]], dtype=np.float64))
+
+    def test_high_blue_no_neutral(self):
+        """Test that high resolution blue with no neutral band can be passed."""
+        from satpy.composites import RatioSharpenedRGB
+        comp = RatioSharpenedRGB(name='true_color', high_resolution_band='blue', neutral_resolution_band=None)
+        res = comp((self.ds1, self.ds2, self.ds3), optional_datasets=(self.ds4,))
+        res = res.values
+        self.assertEqual(res.shape, (3, 2, 2))
+        np.testing.assert_allclose(res[0], np.array([[1.25, 1.25], [np.nan, 0.0]], dtype=np.float64))
+        np.testing.assert_allclose(res[1], np.array([[0.75, 0.75], [np.nan, 0.75]], dtype=np.float64))
+        np.testing.assert_allclose(res[2], np.array([[1.0, 1.0], [np.nan, 1.0]], dtype=np.float64))
+
+    def test_high_blue_neutral_red(self):
+        """Test that high resolution blue with red neutral band can be passed."""
+        from satpy.composites import RatioSharpenedRGB
+        comp = RatioSharpenedRGB(name='true_color', high_resolution_band='blue', neutral_resolution_band='red')
+        res = comp((self.ds1, self.ds2, self.ds3), optional_datasets=(self.ds4,))
+        res = res.values
+        self.assertEqual(res.shape, (3, 2, 2))
+        np.testing.assert_allclose(res[0], np.array([[5.0, 5.0], [np.nan, 0.0]], dtype=np.float64))
+        np.testing.assert_allclose(res[1], np.array([[0.75, 0.75], [np.nan, 0.75]], dtype=np.float64))
+        np.testing.assert_allclose(res[2], np.array([[1.0, 1.0], [np.nan, 1.0]], dtype=np.float64))
+
```

Comment:
Maybe we should convert this class to a pytest-based class (no `unittest.TestCase` subclass), remove the `self.assertX` statements, and then use `@pytest.mark.parametrize` to combine all these tests together. They are all very similar.